hail/src/main/scala/is/hail/services/BatchClient.scala

```diff
@@ -252,6 +255,9 @@ case class BatchClient private (req: Requester) extends Logging with AutoCloseab
         JObject(
           "job_group_id" -> JInt(1), // job group id relative to the update
           "absolute_parent_id" -> JInt(jobGroup.absolute_parent_id),
+          "cancel_after_n_failures" -> jobGroup.cancel_after_n_failures.map(JInt(_)).getOrElse(
+            JNull
+          ),
```

Comment:
thinking about this, `.getOrElse(JNull)` may be unnecessary - you can probably delete it

Reply:
This wont work unless you call `decompose` on the option - ignore the above.